torchreid/utils/tools.py

```diff
@@ -63,15 +63,19 @@ def read_yaml(fpath):
         obj = yaml.safe_load(f)
     return obj
 
-def set_random_seed(seed):
+def set_random_seed(seed, deterministic=False):
     torch.manual_seed(seed)
     torch.cuda.manual_seed_all(seed)
-    torch.backends.cudnn.deterministic = True
-    torch.backends.cudnn.benchmark = False
+    if deterministic:
+        torch.backends.cudnn.deterministic = True
+        torch.backends.cudnn.benchmark = False
     np.random.seed(seed)
     random.seed(seed)
     os.environ['PYTHONHASHSEED'] = str(seed)
 
+def worker_init_fn(worker_id):
+    np.random.seed(np.random.get_state()[1][0] + worker_id)
```

Comment:
I guess, this problem concerns NumPy random generator only according to the article, but I added `random.seed` too

Reply:
The article just provides a sample implementation. Actually the problem concerns all not system-wide random engines since workers are standalone processes.